libs/agno/agno/agent/remote.py

```diff
@@ -0,0 +1,358 @@
+import json
+from dataclasses import dataclass
+from functools import cached_property
+from typing import TYPE_CHECKING, Any, AsyncIterator, Dict, List, Literal, Optional, Sequence, Union, overload
+
+from pydantic import BaseModel
+
+from agno.media import Audio, File, Image, Video
+from agno.models.base import Model
+from agno.models.message import Message
+from agno.models.response import ToolExecution
+from agno.remote.base import BaseRemote, RemoteDb, RemoteKnowledge
+from agno.run.agent import RunOutput, RunOutputEvent
+from agno.utils.agent import validate_input
+from agno.utils.remote import serialize_input
+
+if TYPE_CHECKING:
+    from agno.os.routers.agents.schema import AgentResponse
+
+
+@dataclass
+class RemoteAgent(BaseRemote):
+    def __init__(
+        self,
+        base_url: str,
+        agent_id: str,
+        timeout: float = 60.0,
+    ):
```

Comment:
We could allow for agent_id=None if there's only one in the server. Would give for shortest possible DX